rio_toa/sun_utils.py

```diff
@@ -0,0 +1,196 @@
+import re
+import datetime
+import numpy as np
+
+
+def parse_utc_string(collected_date, collected_time_utc):
+    """
+    Given a string in the format:
+        YYYY-MM-DD HH:MM:SS.SSSSSSSSZ
+    Parse and convert into a datetime object
+    Fractional seconds are ignored
+
+    Parameters
+    -----------
+    collected_date_utc: str
+        Format: YYYY-MM-DD 
+    collected_time: str
+        Format: HH:MM:SS.SSSSSSSSZ
+
+    Returns
+    --------
+    datetime object
+        parsed scene center time
+    """
+    utcstr = collected_date + ' ' + collected_time_utc
+
+    if not re.match(r'[0-9]{4}\-[0-9]{2}\-[0-9]{2}\ [0-9]{2}\:[0-9]{2}\:[0-9]{2}\.[0-9]+Z', utcstr):
+        raise ValueError("%s is an invalid utc time" % utcstr)
+
+    return datetime.datetime.strptime(
+        utcstr.split(".")[0],
+        "%Y-%m-%d %H:%M:%S")
+
+
+def time_to_dec_hour(parsedtime):
+    """
+    Calculate the decimal hour from a datetime object
+
+    Parameters
+    -----------
+    parsedtime: datetime object
+
+    Returns
+    --------
+    decimal hour: float
+        time in decimal hours
+    """
+    return (parsedtime.hour +
+            (parsedtime.minute / 60.0) +
+            (parsedtime.second / 60.0 ** 2)
+            )
+
+
+def calculate_declination(d):
+    """
+    Calculate the declination of the sun in radians based on a given day.
+    As reference +23.26 degrees at the northern summer solstice, -23.26
+    degrees at the southern summer solstice.
+    See: https://en.wikipedia.org/wiki/Position_of_the_Sun#Calculations
+    
+
+    Parameters
+    -----------
+    d: int
+        days from midnight on January 1st
+
+    Returns
+    --------
+    declination in radians: float
+        the declination on day d
+
+    """
+    # return - np.arcsin(0.39799 * np.cos(
+    #             np.deg2rad(0.98565) *
+    #             (d + 10) +
+    #             np.deg2rad(1.914) *
+    #             np.sin(np.deg2rad(0.98565) * (d - 2))))
+    # return np.deg2rad(23.45)*np.sin(np.deg2rad(360)/365.0*(284+d))
```

Comment:
Can we either remove the dead code or include a comment about why it's here (for reference?)